superset/security/manager.py

```diff
@@ -1872,15 +1872,20 @@ def raise_for_access(
                         .one_or_none()
                     )
                     and dashboard_.roles
-                    and (slice_id := form_data.get("slice_id"))
                     and (
-                        slc := self.get_session.query(Slice)
-                        .filter(Slice.id == slice_id)
-                        .one_or_none()
+                        form_data.get("type") == "NATIVE_FILTER"
```

Comment:
I think this needs a few extra validations needed here in order the grant access:
1. We still need to check that the user has access the the dashboard (I'd suggest moving the `self.can_access_dashboard(dashboard_)` check to be right after `and dashboard_.roles` since we want to check dashboard access whether it's via a chart or native filter
2. That the datasource is actually used as a native filter on the dashboard.  We can probably borrow logic from @Vitor-Avila's [PR here](https://github.com/apache/superset/pull/24808/files#diff-bcb3868b9e29aea5f6e87ec641120dae35a86d732501f488a7a6dbdeb450f5c1R2082); performance shouldn't be a concern here since we only need to check one dashboard

Reply:
good catch, @jfrag1!

Reply:
@jfrag1:

1. Oops. Yeah I missed moving `self.can_access_dashboard(dashboard_)` outside of the `or` block. I've placed it at the end of the logic checks as it's likely the most expensive check.
2. That is an option.